app/controllers/users_controller.rb

```diff
@@ -1,13 +1,18 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
-
+  before_action { authorize @user || User }
+  
+  def show
+    
```

Comment:
I noticed in your code, when I click some user profiles, they don't appear and I get an authorization alert. Not sure exactly where this issues is happening but potentially here? I think even private users should still be seen, only their content hidden.